### DISCUSSION TITLE
[New Primitive] `OtpInput`

### DISCUSSION THREAD

**Discussion by brielov** [radix-ui/primitives#3358] [Ideas]
## Feature request
It would be nice to have some sort of pin input for OTPs or email sent short codes for login.

### Overview

Many user authentication strategies rely on sending a one-time password (OTP) or a short code that could be numeric or alphanumeric. You can of course use a simple input for that, but many sites implement one input box for each character which makes for a clear indication and leaves no room for whitespace. It also handles the paste event correctly.

### Examples in other libraries
[Chakra UI](https://chakra-ui.com/docs/components/pin-input)

![image](https://github.com/radix-ui/primitives/assets/42145089/adff3a9d-c4fc-483c-b889-b55ace94e3fb)

### Who does this impact? Who is this for?

Anybody that needs to deal with OTPs

**Comment by sadeghbarati** [comment]
https://github.com/leonardodino/rci

**Comment by pedroapfilho** [comment]
There is an RFC for that, from @benoitgrelard: https://github.com/radix-ui/primitives/pull/1978

**Comment by Jamison1** [comment]
Appears that RFC hasn't had any activity in 9 months

**Comment by jlarmstrongiv** [comment]
Additional examples
- https://park-ui.com/docs/tailwind/components/pin-input
- https://ark-ui.com/docs/components/pin-input
